src/gtclang/Unittest/FileManager.h

```diff
@@ -25,16 +25,24 @@ namespace gtclang {
 /// @brief Handle unittest data files
 /// @ingroup unittest
 class FileManager {
-  std::string dataPath_;
 
 public:
+  enum TestKind { TK_Unittest, TK_Integrationtet };
+
   FileManager();
+  /// @brief set if unit- or integrationtest
+  void setKind(TestKind kind);
 
   /// @brief Get full path of `filename` or abort if file was not found in `dataPath()` directory
   std::string getFile(llvm::StringRef filename) const;
 
   /// @brief Path of the unittest data files
   const std::string& dataPath() const { return dataPath_; }
+
+private:
+  std::string dataPath_;
+
+  TestKind kind_;
```

Comment:
I think the design of the `FileManager` is little odd and needs some improvement. With the introduction of the `TestKind` the whole manager becomes suddenly a state machine and there is no way of telling what `dataPath` returns? There is also no prevention from calling `dataPath()` without first setting the `TestKind`, resulting in undefined behavior. My suggestions would to use a more expressive design:
```
class FileManager {
public:
  /// @brief Get a file relative to the unittest source directory
  static std::string getUnittestFile(llvm::StringRef path);

  /// @brief Get a temporary file relative to the build directory (this will create all missing directories if necessary)
  static std::string createTemporaryFile(llvm::StringRef path);
};
```
which makes the `FileManager` stateless and it's usage less error prone.